### PR TITLE
fix: flaky test `Permissions sets permissions and connect to Dapp`

### LINKED ISSUE #27890: Fix "Permissions sets permissions and connect to Dapp"
### What is this about?

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/105624/workflows/67c7d5c1-32bb-49a8-8c83-1638bae5825a/jobs/3942577/tests

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
We proceed switching to the Extension full view without waiting until the dialog is closed.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27888?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27890

## **Manual testing steps**

1. Check ci

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/dapp-interactions/permissions.spec.js [hunk 1]
```diff
@@ -28,19 +28,15 @@ describe('Permissions', function () {
...
-        await driver.waitUntilXWindowHandles(3);
-        const windowHandles = await driver.getAllWindowHandles();
-        const extension = windowHandles[0];
-        await driver.switchToWindowWithTitle(
-          WINDOW_TITLES.Dialog,
-          windowHandles,
-        );
-        await driver.clickElement({
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        await driver.clickElementAndWaitForWindowToClose({
```
Comment (seaona) [inline test/e2e/tests/dapp-interactions/permissions.spec.js:32]: fixes race condition

File: test/e2e/tests/dapp-interactions/permissions.spec.js [hunk 2]
```diff
@@ -64,21 +60,17 @@ describe('Permissions', function () {
...
 
-        const getAccountsResult = await driver.waitForSelector({
+        await driver.waitForSelector({
           css: '#getAccountsResult',
           text: publicAddress,
         });
-        assert.equal(
-          (await getAccountsResult.getText()).toLowerCase(),
-          publicAddress.toLowerCase(),
-        );
```
Comment (seaona) [inline test/e2e/tests/dapp-interactions/permissions.spec.js:81]: not needed, we find the element by its text above

File: test/e2e/tests/dapp-interactions/permissions.spec.js [hunk 3]
```diff
@@ -28,19 +28,15 @@ describe('Permissions', function () {
           tag: 'button',
         });
 
-        await driver.waitUntilXWindowHandles(3);
-        const windowHandles = await driver.getAllWindowHandles();
-        const extension = windowHandles[0];
-        await driver.switchToWindowWithTitle(
-          WINDOW_TITLES.Dialog,
-          windowHandles,
```
Comment (seaona) [inline test/e2e/tests/dapp-interactions/permissions.spec.js:36]: all this extra logic is not needed. switchToWindowWithTitle already waits for the dialog and switches when its ready

**Review by chloeYue — APPROVED** [review]
LGTM ! Thanks for the fix !

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**